src/main/java/playfriends/mc/plugin/features/playerprofile/SetDiscordEvent.java

```diff
@@ -0,0 +1,34 @@
+package playfriends.mc.plugin.features.playerprofile;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+import playfriends.mc.plugin.api.PlayerEvent;
+
+public class SetDiscordEvent extends Event {
```

Comment:
Can extend `playfriends.mc.plugin.api.PlayerEvent` instead of `Event`.

Reply:
Should it? (i'm assuming yes now that you've commented on it)

I saw that and thought it would be weird to create a whole inheritance chain where each subclass adds a single extra property.
But in hindsight, that might be because i've been doing a lot of typescript recently, and types there are structural and don't care about inheritance, making readability the only concern

Will change

Reply:
My reasoning for it is twofold:
- I like the indication that this is a `PlayerEvent`. Spigot seems to have a fairly large event hierarchy and actually already has a `PlayerEvent`, but sadly it's not public so I just created a copy here.
- It's a convenience to not have to duplicate the fields/getters (and avoid inconsistency with getter or fieldnames). There's negligable overhead to doing this vs duplicating code (though I admit it's not much 🤷), as all Java's methods are virtual anyway.